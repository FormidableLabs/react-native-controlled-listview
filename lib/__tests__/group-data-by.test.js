/*global it, expect*/
import 'react-native';
import groupDataBy from '../group-data-by';
import people from './__data__/people';

it('groups items based on the provided function', () => {
  expect(groupDataBy((person) => person.lastName[0], people)).toMatchSnapshot();
});
