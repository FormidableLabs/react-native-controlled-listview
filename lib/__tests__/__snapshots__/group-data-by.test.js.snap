exports[`test groups items based on the provided function 1`] = `
Object {
  "B": Array [
    Object {
      "firstName": "Mildred",
      "lastName": "Bonk",
    },
  ],
  "G": Array [
    Object {
      "firstName": "Don",
      "lastName": "Gately",
    },
  ],
  "I": Array [
    Object {
      "firstName": "Orin",
      "lastName": "Incandenza",
    },
    Object {
      "firstName": "Avril",
      "lastName": "Incandenza",
    },
    Object {
      "firstName": "Hal",
      "lastName": "Incandenza",
    },
  ],
  "P": Array [
    Object {
      "firstName": "Michael",
      "lastName": "Pemulis",
    },
  ],
}
`;
